File: app/src/main/java/co/cropbit/sahathanahomecare/ProfileActivity.java
package co.cropbit.sahathanahomecare;

import android.support.annotation.IdRes;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import org.w3c.dom.Text;

public class ProfileActivity extends AppCompatActivity {

    private FirebaseDatabase mDatabase;
    private FirebaseAuth mAuth;
    private FirebaseUser currentUser;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mDatabase = FirebaseDatabase.getInstance();
        mAuth = FirebaseAuth.getInstance();
        currentUser = mAuth.getCurrentUser();
        setContentView(R.layout.activity_profile);
    }

    @Override
    protected void onStart() {
        super.onStart();
        if(currentUser != null) {
            getTextView(R.id.profile_email).setText(currentUser.getEmail());
            DatabaseReference ref = mDatabase.getReference("user").child(currentUser.getUid());
            DatabaseReference nameRef = ref.child("name");
            DatabaseReference phnoRef = ref.child("phno");
            nameRef.addValueEventListener(new ValueEventListener() {
                @Override
                public void onDataChange(DataSnapshot dataSnapshot) {
                    getTextView(R.id.profile_name).setText(dataSnapshot.getValue(String.class));
                }

                @Override
                public void onCancelled(DatabaseError databaseError) {

                }
            });
            phnoRef.addValueEventListener(new ValueEventListener() {
                @Override
                public void onDataChange(DataSnapshot dataSnapshot) {
                    getTextView(R.id.profile_phno).setText(dataSnapshot.getValue(String.class));
                }

                @Override
                public void onCancelled(DatabaseError databaseError) {

                }
            });
        }
    }

    private TextView getTextView(@IdRes int id) {
        return ((TextView) findViewById(id));
    }

}

File: app/src/main/java/co/cropbit/sahathanahomecare/ProfileActivity.kt
package co.cropbit.sahathanahomecare

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import android.view.View
import co.cropbit.sahathanahomecare.model.Hospital
import co.cropbit.sahathanahomecare.model.User
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*
import kotlinx.android.synthetic.main.activity_profile.*

class ProfileActivity : AppCompatActivity() {

    val mAuth: FirebaseAuth = FirebaseAuth.getInstance()
    val mDatabase: FirebaseDatabase = FirebaseDatabase.getInstance()
    var user: User? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_profile)
        setSupportActionBar(toolbarProfile)
        supportActionBar?.setDisplayShowTitleEnabled(true)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        mDatabase.getReference("users").child(mAuth.currentUser?.uid).addValueEventListener(object: ValueEventListener {
            override fun onDataChange(p0: DataSnapshot?) {
                user = p0?.getValue(User::class.java)
                render(user)
            }
            override fun onCancelled(p0: DatabaseError?) {
                user = null
                render(user)
            }
        })
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when(item?.itemId) {
            android.R.id.home -> onBackPressed()
        }
        return super.onOptionsItemSelected(item)
    }

    fun logOut(view: View?) {
        mAuth.signOut()
    }

    fun gotoHistory(view: View?) {
        val intent = Intent(this, HistoryActivity::class.java)
        startActivity(intent)
    }

    fun render(user: User?) {
        profileName.setText(user?.name)
        profilePhno.setText(user?.phoneNumber)
        getHospitals()
    }

    fun getHospitals() {
        mDatabase.getReference("hospitals").addChildEventListener(object: ChildEventListener {
            override fun onChildAdded(p0: DataSnapshot?, p1: String?) {
                var hospital = p0?.getValue(Hospital::class.java)
                Log.v("Sahathana", hospital?.address)
            }

            override fun onChildChanged(p0: DataSnapshot?, p1: String?) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }

            override fun onChildMoved(p0: DataSnapshot?, p1: String?) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }

            override fun onChildRemoved(p0: DataSnapshot?) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }

            override fun onCancelled(p0: DatabaseError?) {
                Log.v("Sahathana", p0?.message)
            }
        })
    }
}
